validator-cli/src/ArbToEth/transactionHandler.ts

```diff
@@ -0,0 +1,191 @@
+type Transactions = {
+  challengeTxn: string | null;
+  withdrawChallengeDepositTxn: string | null;
+  sendSnapshotTxn: string | null;
+  executeSnapshotTxn: string | null;
+};
+
+export enum ContractType {
+  INBOX = "inbox",
+  OUTBOX = "outbox",
+}
+
+export class ArbToEthTransactionHandler {
+  public requiredConfirmations = 10;
+  public claim: ClaimStruct | null = null;
+  public chainId = 11155111;
+
+  public veaInbox: VeaInboxArbToEth;
+  public veaOutbox: VeaOutboxArbToEth;
+  public veaInboxProvider: JsonRpcProvider;
+  public veaOutboxProvider: JsonRpcProvider;
+  public epoch: number;
+  public emitter: typeof defaultEmitter;
+
+  public transactions: Transactions = {
+    challengeTxn: null,
+    withdrawChallengeDepositTxn: null,
+    sendSnapshotTxn: null,
+    executeSnapshotTxn: null,
+  };
+
+  constructor(
+    epoch: number,
+    veaInbox: VeaInboxArbToEth,
+    veaOutbox: VeaOutboxArbToEth,
+    veaInboxProvider: JsonRpcProvider,
+    veaOutboxProvider: JsonRpcProvider,
+    emitter: typeof defaultEmitter = defaultEmitter,
+    claim: ClaimStruct | null = null
+  ) {
+    this.epoch = epoch;
+    this.veaInbox = veaInbox;
+    this.veaOutbox = veaOutbox;
+    this.veaInboxProvider = veaInboxProvider;
+    this.veaOutboxProvider = veaOutboxProvider;
+    this.emitter = emitter;
+    this.claim = claim;
+  }
+
+  /**
+   * Check the status of a transaction.
+   *
+   * @param trnxHash Transaction hash to check the status of.
+   * @param contract Contract type to check the transaction status in.
+   *
+   * @returns False if transaction is pending || not final || not made, else True.
+   */
+  public async checkTransactionStatus(trnxHash: string | null, contract: ContractType): Promise<number> {
+    let provider: JsonRpcProvider;
+    if (contract === ContractType.INBOX) {
+      provider = this.veaInboxProvider;
+    } else if (contract === ContractType.OUTBOX) {
+      provider = this.veaOutboxProvider;
+    }
+
+    if (trnxHash == null) {
+      return 0;
+    }
+
+    const receipt = await provider.getTransactionReceipt(trnxHash);
+
+    if (!receipt) {
+      // TODO: Add transaction pending timeout- redo transaction.
+      this.emitter.emit(BotEvents.TXN_PENDING, trnxHash);
+      return 1;
+    }
+
+    const currentBlock = await provider.getBlock("latest");
+    const confirmations = currentBlock.number - receipt.blockNumber;
+
+    if (confirmations >= this.requiredConfirmations) {
+      this.emitter.emit(BotEvents.TXN_FINAL, trnxHash, confirmations);
+      return 3;
+    } else {
+      this.emitter.emit(BotEvents.TXN_NOT_FINAL, trnxHash, confirmations);
+      return 2;
+    }
+  }
+
+  /**
+   * Challenge claim for this.epoch in VeaOutbox(ETH).
+   *
+   */
+  public async challengeClaim() {
+    this.emitter.emit(BotEvents.CHALLENGING);
+    if (!this.claim) {
+      throw new ClaimNotSetError();
+    }
+    if ((await this.checkTransactionStatus(this.transactions.challengeTxn, ContractType.OUTBOX)) > 0) {
```

Comment:
(ignoreable) Consider extracting the await statement to a named variable for readability